### PR TITLE
Save crawler run metadata and log file in a run-specific folder

### PR DESCRIPTION
**Pull request by elic-eon**
### 🤔 Why?

To capture each run metadata and log

### 🤓 What?

- Sink all output file in a run-specific folder

### 🧪 Tested?

Updated tests and verified locally.

### CODE REVIEW

File: metaphor/common/file_sink.py [hunk 1]
```diff
@@ -50,6 +50,7 @@ def __init__(self, config: FileSinkConfig):
         self.path = config.directory.rstrip("/")
         self.batch_size = config.batch_size
         self.write_logs = config.write_logs
+        self.run_dir = datetime.now(timezone.utc).strftime("%Y-%m-%d %H-%M-%S")
```
Comment (mars-lan) [inline metaphor/common/file_sink.py:53]: Let's just use timestamp to avoid any potential issues with spaces in the S3 path?

Comment (elic-eon) [reply metaphor/common/file_sink.py]: Do you mean the unix timestamp, a number?
Changed to `%Y-%m-%d-%H%M%S` for now

Comment (mars-lan) [reply metaphor/common/file_sink.py]: Correct, e.g. `s3://crawler_x/1673544307/run.metadata`

Comment (elic-eon) [reply metaphor/common/file_sink.py]: Why not use a readable timestamp without space or other character?

**Review by mars-lan — APPROVED** [review]

**Pull request merged**